sdk/rust/nym-sdk/src/mixnet/native_client.rs

```diff
@@ -190,6 +214,44 @@ impl MixnetClient {
         // note: it's important to take ownership of the struct as if the shutdown is `TaskHandle::External`,
         // it must be dropped to finalize the shutdown
     }
+
+    // fn read_buffer_to_slice(
+    //     &mut self,
+    //     buf: &mut [u8],
+    //     cx: &mut Context<'_>,
+    // ) -> Poll<std::result::Result<usize, std::io::Error>> {
+    //     if self._read.buffer.len() < buf.len() {
+    //         let written = self._read.buffer.len();
+    //         buf[..written].copy_from_slice(&self._read.buffer);
+    //         self._read.clear();
+    //         Poll::Ready(Ok(written))
+    //     } else {
+    //         let written = buf.len();
+    //         buf.copy_from_slice(&self._read.buffer[..written]);
+    //         self._read.buffer = self._read.buffer[written..].to_vec();
+    //         cx.waker().wake_by_ref();
+    //         Poll::Ready(Ok(written))
+    //     }
+    // }
+
+    fn read_buffer_to_slice(
+        &mut self,
+        buf: &mut ReadBuf,
+        cx: &mut Context<'_>,
+    ) -> Poll<tokio::io::Result<()>> {
+        if self._read.buffer.len() < buf.capacity() {
+            // let written = self._read.buffer.len();
+            buf.put_slice(&self._read.buffer);
+            self._read.clear();
+            Poll::Ready(Ok(()))
+        } else {
+            let written = buf.capacity();
+            buf.put_slice(&self._read.buffer[..written]);
```

Comment:
nit: you could instead use `buf.put_slice(&self_read.buffer.split_off(written))` then you wouldn't need to worry about manually advancing the buffer